geniepy/src/geniepy/datamgmt/scrapers.py

```diff
@@ -148,6 +151,118 @@ def scrape(self, chunksize: int, **kwargs) -> Generator:
         raise NotImplementedError
 
 
+class CitationScraper(BaseScraper):
+    """
+    Implementation of PubMed Articles Scraper.
+
+    https://www.ncbi.nlm.nih.gov/pubmed/
+    https://www.nlm.nih.gov/bsd/licensee/elements_descriptions.html
+
+    """
+
+    # Constants
+    API_BASE_URL = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&linkname=pubmed_pubmed_citedin"  # noqa
+    API_ID_PARAM = "&id="
+    API_KEY_PARAM = "&api_key="
+    API_KEYS = ["70b53d4e84436970587aef3493a723cae708",
+                "9ab4b04dcabcab740d1a297e6f3f54aa1e09",
+                "39458b83e654b32523475ee6b828b6c22c08",
+                "cf95d471b41c4ea9e6b89d41de269705eb08",
+                "94c3049a7b3dffca2d22ce11b51a3ea5da09",
+                "3fb37654e8f711f8751d9e3c21fc46257108",
+                "91315e4e69c1b5e2676371490cfc715caa08",
+                "b1ee76fae01c312e4bf2fd3e8b1bcf978a09",
+                "25b74bb456d7e6fc756c86c7aebb1b525009",
+                "7a7e4492084dd09cef758ee65a7909704b08",
+                "4d32405f13fed69766ad639d89c5c5288608",
+                "18c8d26e60b5cd012e2ba9618c88dd12cb09",
+                "667ca5b1b72b7697ed43c26727329282bb08",
+                "5cdea4b682e2fc93c56c2b1e5e9da93f4709",
+                "108b2e1a81369991da6c737545097c581a08",
+                "f10805158a7609bb115cb074b05e5923a407",
+                "00d74dd1cb732cd7fc29f54168fe7055c809"]
+    TAG_CITATION_ID = "./LinkSet/LinkSetDb/Link/Id"                   
+
+    def scrape(self, chunksize: int, **kwargs) -> Generator:
+        """
+        Implement base scrape method to download records from PubMed database.
+
+        Keyword Arguments:
+            chunksize {int} -- the size of each chunk the data should be returned
+
+        Returns:
+            Generator -- The generator yielding the data in given chunksizes.
+        """
+
+        chunksize_min: int = 1
+        chunksize_max: int = 100
```

Comment:
Why are you limiting chunksize to 100? The default in the config file is 100k, so this fails straight away. I'll only make the lower boundary check - Afterwards it would be a good idea to make max_chunksize part of the config module